File: app/src/main/java/org/cuieney/videolife/presenter/VideoHomePresenter.java
package org.cuieney.videolife.presenter;

import org.cuieney.videolife.common.api.KyApiService;
import org.cuieney.videolife.common.base.RxPresenter;
import org.cuieney.videolife.common.utils.RxUtil;
import org.cuieney.videolife.presenter.contract.VideoHomeContract;

import javax.inject.Inject;

import rx.Subscription;

/**
 * Created by cuieney on 17/2/24.
 */

public class VideoHomePresenter extends RxPresenter<VideoHomeContract.View> implements VideoHomeContract.Presenter{

    private KyApiService mRetrofitHelper;

    @Inject
    public VideoHomePresenter(KyApiService mRetrofitHelper) {
        this.mRetrofitHelper = mRetrofitHelper;
    }

    @Override
    public void getVideoData(String date) {
        Subscription rxSubscription = mRetrofitHelper.getVideoList(date)
                .compose(RxUtil.rxSchedulerHelper())
                .subscribe(videoListBean -> {
                    mView.showContent(videoListBean);
                }, throwable -> {
                    mView.error(throwable);
                });
        addSubscrebe(rxSubscription);
    }
}

File: app/src/main/java/org/cuieney/videolife/kotlin/presenter/VideoHomePresenter.kt
package org.cuieney.videolife.kotlin.presenter

import org.cuieney.videolife.common.api.KyApiService
import org.cuieney.videolife.common.utils.RxUtil
import org.cuieney.videolife.entity.VideoListBean
import org.cuieney.videolife.kotlin.base.RxPresenter
import org.cuieney.videolife.kotlin.presenter.contract.VideoHomeContract
import javax.inject.Inject

/**
 * Created by cuieney on 2017/6/2.
 */
class VideoHomePresenter
@Inject constructor(val retrofitHelper: KyApiService) : RxPresenter(), VideoHomeContract.Presenter {

    override fun getVideoData(date: String, view: VideoHomeContract.View) {
        val rxSubscription = retrofitHelper.getVideoList(date)
                .compose<VideoListBean>(RxUtil.rxSchedulerHelper<VideoListBean>())
                .subscribe({ t -> view.showContent(t) }) { throwable -> view.error(throwable) }
        addSubscrebe(rxSubscription)
    }
}